app/permission/foreground/main.go

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+//+build !android
+
+/*
+Package foreground implements permissions to run a foreground service.
+See https://developer.android.com/guide/components/foreground-services.
+
+The following entries will be added to AndroidManifest.xml:
+
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+
+*/
+
+package foreground
+
+// StartBackground notifies the system that the program will perform
+// background work and that it shouldn't be killed. It returns a channel
+// that should be closed when the background work is complete.
+
+// StartBackground is a no-op on Linux, Windows, macOS; Android will
+// display a notification during background work; iOS isn't supported.
+func StartForeground() chan struct{} {
```

Comment:
Repeating the documentation (and function signature) can be avoided by introducing an unexported startForeground function which is platform specific, and remove build constraints for Start and its documentation.